File: dotnet/source/cmf.bus/IEnvelopeSender.cs
using System;

namespace cmf.bus
{
    /// <summary>
    /// An interface defining the methods by which an client may send envelopes.
    /// </summary>
    public interface IEnvelopeSender : IDisposable
    {
        /// <summary>
        /// Sends an envelope.
        /// </summary>
        /// <param name="env">The envelope to be sent</param>
        /// <exception cref="System.Exception">May throw an exception.</exception>
        void Send(Envelope env);
    }
}

File: java/cmf-bus-api/src/main/java/cmf/bus/IEnvelopeSender.java
package cmf.bus;

/**
 * An interface defining the methods by which an client may send envelopes.
 * @see IEnvelopeBus
 */
public interface IEnvelopeSender {

	/**
	 * Sends an envelope.
	 * 
	 * @param envelope The envelope to be sent.
	 * 
	 * @throws Exception
	 */
    void send(Envelope envelope) throws Exception;
}
